### PR TITLE
Removed SmartEnum2

### PR DESCRIPTION
**Pull request by mimse**
## Description

Removed SmartEnum to be able to reuse enums across projects

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Domain/Energinet.DataHub.MarketParticipant.Domain.csproj [hunk 1]
```diff
@@ -24,8 +24,7 @@ limitations under the License.
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Ardalis.SmartEnum" Version="2.1.0" />
-    <PackageReference Include="Ardalis.SmartEnum.SystemTextJson" Version="2.1.0" />
+    <ProjectReference Include="..\Energinet.DataHub.MarketParticipant.Client\source\Energinet.DataHub.MarketParticipant.Client\Energinet.DataHub.MarketParticipant.Client.csproj" />
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Energinet.DataHub.MarketParticipant.Domain.csproj:27]: Keep enums in domain, do not reference client package here.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/ContactCategory.cs [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-using System.Runtime.CompilerServices;
-using Ardalis.SmartEnum;
-
-namespace Energinet.DataHub.MarketParticipant.Domain.Model
-{
-    public sealed class ContactCategory : SmartEnum<ContactCategory>
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/ContactCategory.cs:20]: Make regular enum.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeAddedToActorIntegrationEvent.cs [hunk 3]
```diff
@@ -24,6 +24,6 @@ public sealed class MeteringPointTypeAddedToActorIntegrationEvent : IntegrationE
     public Guid ActorId { get; set; }
     public EicFunction Function { get; set; }
     public Guid GridAreaId { get; set; }
-    [JsonConverter(typeof(SmartEnumNameConverter<MeteringPointType, int>))]
+    [JsonConverter(typeof(MeteringPointType))]
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeAddedToActorIntegrationEvent.cs:27]: JsonConverter is afaik configured already. Should be possible to remove attribute.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/MeteringPointType.cs [hunk 4]
```diff
@@ -1,101 +0,0 @@
...
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-using System.Runtime.CompilerServices;
-using Ardalis.SmartEnum;
-
-namespace Energinet.DataHub.MarketParticipant.Domain.Model
-{
-    public sealed class MeteringPointType : SmartEnum<MeteringPointType>
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/MeteringPointType.cs:20]: Make regular enum.

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs [hunk 5]
```diff
@@ -64,7 +64,7 @@ public Task EnqueueActorUpdatedEventAsync(OrganizationId organizationId, Actor a
                         actorMarketRole.GridAreas.Select(
                             x => new ActorGridAreaEventData(
                                 x.Id,
-                                x.MeteringPointTypes.Select(y => y.Name).ToList()))
+                                x.MeteringPointTypes.Select(y => y.ToString()).ToList()))
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs:67]: Change type from string[] to MeteringPointType[].

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs [hunk 6]
```diff
@@ -99,7 +99,7 @@ public Task EnqueueActorCreatedEventsAsync(OrganizationId organizationId, Actor
                         actorMarketRole.GridAreas.Select(
                                 x => new ActorGridAreaEventData(
                                     x.Id,
-                                    x.MeteringPointTypes.Select(y => y.Name).ToList()))
+                                    x.MeteringPointTypes.Select(y => y.ToString()).ToList()))
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs:102]: Change type from string[] to MeteringPointType[].

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/EntityConfiguration/ActorContactEntityConfiguration.cs [hunk 7]
```diff
@@ -29,9 +28,6 @@ public void Configure(EntityTypeBuilder<ActorContactEntity> builder)
             builder.HasKey(contact => contact.Id);
             builder.Property(contact => contact.Id).ValueGeneratedOnAdd();
             builder.Property(contact => contact.Category)
-                .HasConversion(
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/EntityConfiguration/ActorContactEntityConfiguration.cs:32]: You still need to convert to and from enum.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/MeteringPointType.cs (hunk 4 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/MeteringPointType.cs:20]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/a31a24627f75fea737c871171eb54949e8cc598c

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/ContactCategory.cs (hunk 2 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/ContactCategory.cs:20]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/a31a24627f75fea737c871171eb54949e8cc598c

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/ActorContactEventData.cs [hunk 8]
```diff
@@ -29,7 +29,7 @@ public ActorContactEventData(string name, EmailAddress email, ContactCategory ca
 
         public string Name { get; }
         public EmailAddress Email { get; }
-        [JsonConverter(typeof(SmartEnumNameConverter<ContactCategory, int>))]
+        [JsonConverter(typeof(ContactCategory))]
```
[the remark this answers is not in the document]
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/ActorContactEventData.cs]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/a31a24627f75fea737c871171eb54949e8cc598c

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeAddedToActorIntegrationEvent.cs (hunk 3 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeAddedToActorIntegrationEvent.cs]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/a31a24627f75fea737c871171eb54949e8cc598c

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeRemovedFromActorIntegrationEvent.cs [hunk 9]
```diff
@@ -24,6 +24,6 @@ public sealed class MeteringPointTypeRemovedFromActorIntegrationEvent : Integrat
     public Guid ActorId { get; set; }
     public EicFunction Function { get; set; }
     public Guid GridAreaId { get; set; }
-    [JsonConverter(typeof(SmartEnumNameConverter<MeteringPointType, int>))]
+    [JsonConverter(typeof(MeteringPointType))]
```
[the remark this answers is not in the document]
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/IntegrationEvents/ActorIntegrationEvents/MeteringPointTypeRemovedFromActorIntegrationEvent.cs]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/a31a24627f75fea737c871171eb54949e8cc598c

File: source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/EntityConfiguration/ActorContactEntityConfiguration.cs (hunk 7 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Infrastructure/Persistence/EntityConfiguration/ActorContactEntityConfiguration.cs:32]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/ff6cd7c8269908aa2f8f6d279f22a2e3a11e7cf4

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs (hunk 6 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/ff6cd7c8269908aa2f8f6d279f22a2e3a11e7cf4

File: source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs (hunk 5 above)
Comment (mimse) [reply source/Energinet.DataHub.MarketParticipant.Domain/Services/ActorIntegrationEventsQueueService.cs]: Solved in https://github.com/Energinet-DataHub/geh-market-participant/pull/283/commits/ff6cd7c8269908aa2f8f6d279f22a2e3a11e7cf4

**Review by defectiveAi — APPROVED** [review]
🎆

**Pull request merged**